psi4/src/psi4/libmints/mintshelper.cc

```diff
@@ -2682,6 +2682,61 @@ std::vector<SharedMatrix> MintsHelper::ao_overlap_kinetic_deriv2_helper(const st
     return grad;
 }
 
+/* 1st derivatives of electric dipole integrals in the AO basis */
+std::vector<SharedMatrix> MintsHelper::ao_elec_dip_deriv1_helper(int atom) {
+    std::vector<std::string> cartcomp{ "X", "Y", "Z" };
+
+    std::shared_ptr<OneBodyAOInt> Dint(integral_->ao_dipole(1));
+
+    std::shared_ptr<BasisSet> bs1 = Dint->basis1();
+    std::shared_ptr<BasisSet> bs2 = Dint->basis2();
+
+    int nbf1 = bs1->nbf();
+    int nbf2 = bs2->nbf();
+
+    std::vector<SharedMatrix> grad;
+    for (int p = 0; p < 3; p++) {
+        std::stringstream sstream;
+        sstream << "ao_mu" << cartcomp[p] << "_deriv1_";
+        for (int q = 0; q < 3; q++) {
+            sstream << atom << cartcomp[q];
+            grad.push_back(SharedMatrix(new Matrix(sstream.str(), nbf1, nbf2)));
+        }
+    }
+
+    const double *buffer = Dint->buffer();
+
+    for (int P = 0; P < bs1->nshell(); P++) {
+        for (int Q = 0; Q < bs2->nshell(); Q++) {
+            int nP = basisset_->shell(P).nfunction();
+            int oP = basisset_->shell(P).function_index();
+            int aP = basisset_->shell(P).ncenter();
+
+            int nQ = basisset_->shell(Q).nfunction();
+            int oQ = basisset_->shell(Q).function_index();
+            int aQ = basisset_->shell(Q).ncenter();
+
+            if (aP != atom && aQ != atom) continue;
```

Comment:
Is this right, the atoms of P and Q have to be the same?

Reply:
Not exactly. This line checks to make sure that the atom passed is equal to at least one of them before computing any derivatives

Reply:
Doh, you're right.